### PR TITLE
[AIRFLOW-6360] 'Recent tasks' stats only show non-completed dagruns o…

### PR DESCRIPTION
**Pull request by tooptoop4**
…ption

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [AIRFLOW-6360](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-6360\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6360
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-6360\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by drsantos20 — COMMENTED** [review]
Hi @tooptoop4 do we already have tests for these changes?

**Comment by tooptoop4** [comment]
@drsantos20 i couldn't see a test for original task_stats functionality, as this change is guarded by opt-in config then no impact for existing users

**Comment by potiuk** [comment]
@tooptoop4  -> Still the tests are targeting the "future" rather than the past.

We often cherry-pick some changes to 1.10 now + we want to make sure that we can refactor Airflow in the future without having to run tests manually, so we are now fairly strict in requiring tests for new functionalities - even if there were missing tests in the past. If you want to add new functionality, we expect you add tests, sorry :(.

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -377,6 +377,9 @@ update_fab_perms = True
 # 0 means never get forcibly logged out
 force_log_out_after = 0
 
+# 'Recent Tasks' stats will show for old DagRuns if True
```
Comment (mik-laj) [inline airflow/config_templates/default_airflow.cfg:380]: Hi. Can you add some documentation?  We do not yet have a document describing the configuration options of the webserver. Would you like to create it? If not, that's fine and you can also create a new document in the howto directory.

Comment (tooptoop4) [reply airflow/config_templates/default_airflow.cfg:380]: see https://github.com/apache/airflow/pull/7037

**Comment by tooptoop4** [comment]
superseded by https://github.com/apache/airflow/pull/7037

**Pull request closed without merging**